f4se-plugin/src/Menu/NAFMenu/MenuHandlers/InventoryMenuHandler.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include "Menu/BindableMenu.h"
+#include "Misc/MathUtil.h"
+#include "Scene/SceneBase.h"
+#include "Scene/SceneManager.h"
+
+namespace Menu::NAF
+{
+	class InventoryMenuHandler : public BindableMenu<InventoryMenuHandler, SUB_MENU_TYPE::kInventories>
+	{
+	public:
+		using BindableMenu::BindableMenu;
+
+		uint64_t curSceneId = 0;
+
+		virtual ~InventoryMenuHandler() {}
+
+		virtual void InitSubmenu() override
+		{
+			BindableMenu::InitSubmenu();
+			auto sceneData = PersistentMenuState::SceneData::GetSingleton();
+			curSceneId = sceneData->pendingSceneId;
+			sceneData->pendingSceneId = 0;
+		}
+
+		virtual BindingsVector GetBindings() override
+		{			
+			BindingsVector result;
+			Scene::SceneManager::VisitScene(curSceneId, [&](Scene::IScene* scn) {
+				scn->ForEachActor([&](RE::Actor* currentActor, Scene::ActorPropertyMap&) {
+					result.push_back({ GameUtil::GetActorName(currentActor),
+						Bind(&InventoryMenuHandler::ShowActorInventory, currentActor->GetActorHandle()) });
+				});
+			});
+
+			return result;
+		}
+
+		void ShowActorInventory(RE::ActorHandle h, int)
+		{
+			RE::Actor* a = h.get().get();
+			if (a)
+			{
+				bool hadSWIKeyword = a->HasKeyword(Data::Forms::ShowWornItemsKW);
+				if (!hadSWIKeyword)
+				{
+					a->ModifyKeyword(Data::Forms::ShowWornItemsKW, true);
+				}
+
+				Menu::IStateManager::activeInstance->CloseMenu();
+				Sleep(100);
```

Comment:
Is it necessary to suspend thread execution?

Reply:
yes, but it probably can be done in a more elegant way.

Reply:
actually, after the commit to unpause the containermenu, the delay is no longer necessary.
I have removed it.